tracer/test/Datadog.Trace.Tests/TraceContextTests.cs

```diff
@@ -195,15 +195,87 @@ public void PartialFlushShouldPropagateMetadata(bool inAASContext)
             tracer.Setup(t => t.Write(It.IsAny<ArraySegment<Span>>()))
                   .Callback<ArraySegment<Span>>(s => spans = s);
 
-            var vars = Environment.GetEnvironmentVariables();
+            SetAASContext(inAASContext);
+            var traceContext = new TraceContext(tracer.Object);
+            traceContext.SetSamplingPriority(SamplingPriorityValues.UserKeep);
+
+            var rootSpan = CreateSpan();
+
+            traceContext.AddSpan(rootSpan);
+
+            for (int i = 0; i < partialFlushThreshold - 1; i++)
+            {
+                var span = CreateSpan();
+                traceContext.AddSpan(span);
+                traceContext.CloseSpan(span);
+            }
+
+            // At this point, only one span is missing to reach the threshold for partial flush
+            spans.Should().BeNull("partial flush should not have been triggered");
+
+            // Closing the root span brings the number of closed spans to the threshold
+            // but a full flush should be triggered rather than a partial, because every span in the trace has been closed
+            traceContext.CloseSpan(rootSpan);
+
+            spans.Value.Should().NotBeNullOrEmpty("a full flush should have been triggered");
+
+            rootSpan.GetMetric(Metrics.SamplingPriority).Should().Be(SamplingPriorityValues.UserKeep, "priority should be assigned to the root span");
+
+            spans.Value.Should().OnlyContain(s => s == rootSpan || s.GetMetric(Metrics.SamplingPriority) == null, "only the root span should have a priority");
+
+            if (inAASContext)
+            {
+                spans.Value.Array[0].GetTag(Tags.AzureAppServicesResourceGroup).Should().NotBeNull();
+            }
+            else
+            {
+                spans.Value.Array[0].GetTag(Tags.AzureAppServicesResourceGroup).Should().BeNull();
+            }
+
+            // Now test the case where a spans gets opened when the root has been sent (It can happen)
```

Comment:
Maybe I'm misunderstanding, but this doesn't appear to be testing this scenario? 🤔

Reply:
Hmm, I believe it does. As above, all spans on the context have been closed, thus the spans registered before are sent. So no local spans are in the context. Then when we add and close a new one, we are in a context where the first chunk has already been set then we need to add the extra aas metadata.